runtime/src/accounts.rs

```diff
@@ -151,50 +151,65 @@ impl Accounts {
             // If a fee can pay for execution then the program will be scheduled
-            let mut accounts: TransactionAccounts = Vec::with_capacity(message.account_keys.len());
+            let mut payer_index = None;
             let mut tx_rent: TransactionRent = 0;
-            for (i, key) in message
+            let mut accounts: Vec<_> = message
                 .account_keys
                 .iter()
                 .enumerate()
-                .filter(|(i, key)| Self::is_non_loader_key(message, key, *i))
-            {
-                let (account, rent) = AccountsDB::load(storage, ancestors, accounts_index, key)
-                    .and_then(|(mut account, _)| {
-                        if message.is_writable(i) && !account.executable {
-                            let rent_due = rent_collector.update(&key, &mut account);
-                            Some((account, rent_due))
-                        } else {
-                            Some((account, 0))
+                .map(|(i, key)| {
+                    if Self::is_non_loader_key(message, key, i) {
+                        if payer_index.is_none() {
+                            payer_index = Some(i);
                         }
-                    })
-                    .unwrap_or_default();
-
-                accounts.push(account);
-                tx_rent += rent;
-            }
-
-            if accounts.is_empty() || accounts[0].lamports == 0 {
-                error_counters.account_not_found += 1;
-                Err(TransactionError::AccountNotFound)
-            } else {
-                let min_balance = match get_system_account_kind(&accounts[0]).ok_or_else(|| {
-                    error_counters.invalid_account_for_fee += 1;
-                    TransactionError::InvalidAccountForFee
-                })? {
-                    SystemAccountKind::System => 0,
-                    SystemAccountKind::Nonce => {
-                        rent_collector.rent.minimum_balance(nonce::State::size())
+                        let (account, rent) =
+                            AccountsDB::load(storage, ancestors, accounts_index, key)
+                                .and_then(|(mut account, _)| {
+                                    if message.is_writable(i) && !account.executable {
+                                        let rent_due = rent_collector.update(&key, &mut account);
+                                        Some((account, rent_due))
+                                    } else {
+                                        Some((account, 0))
+                                    }
+                                })
+                                .unwrap_or_default();
+
+                        tx_rent += rent;
+                        account
+                    } else {
+                        // Fill in an empty account for the program slots.
+                        Account::default()
```

Comment:
@jackcmay do you feel that we need to actually load the program accounts?

Reply:
No, don't need to, default accounts are probaby ok for now